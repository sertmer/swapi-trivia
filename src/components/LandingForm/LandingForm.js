import React, { Component } from 'react'
import './LandingForm.scss'
import { Link } from 'react-router-dom'

class LandingForm extends Component {
  constructor(props) {
    super(props)
    this.state = {
      name: '',
      quote: '',
      rank: '',
      isFinished: false
    }
  }

  handleChange = (e) => {
    this.setState({[e.target.name]: e.target.value})
    const isNameGood = this.state.name !== '';
    const isQuoteGood = this.state.quote !== '';
    const isRankGood = this.state.rank !== 'rank';

    if(isNameGood && isQuoteGood && isRankGood) {
      this.setState({ isFinished: true })
    } else {
      this.setState({ isFinished: false })
    };
  }

  submitInfo = () => {
    let name = this.state.name
    let quote = this.state.quote
    let rank = this.state.rank
    this.props.updateState({name, quote, rank});
  }

  render() {
    return (
      <div>
        <input 
          className='form_name' 
          type='text' 
          placeholder='Name' 
          value={this.state.name}
          name='name'
          onChange={this.handleChange}
          />
        <input 
          className='form_quote'
          type='text' 
          placeholder='Favorite Star Wars Quote' 
          value={this.state.quote}
          name='quote'
          onChange={this.handleChange}
        /> 
        <select className='rank-menu' name="rank" value={this.state.rank} onChange={this.handleChange}>
          <option defaultValue='rank'>Select Your Rank..</option>
          <option name='rank' value='Padawan'>Padawan</option>
          <option name='rank' value='Jedi Knight'>Jedi Knight</option>
          <option name='rank' value='Jedi Master'>Jedi Master</option>
        </select>
        {!this.state.isFinished && <button 
            className="form_button" 
            disabled  
          >Submit</button>}
          {this.state.isFinished && <Link to='/movies'>
          <button
            className="form_button"
            onClick={this.submitInfo}
          >Submit</button>
            </Link>}
      </div>
    )
  }
}

export default LandingForm