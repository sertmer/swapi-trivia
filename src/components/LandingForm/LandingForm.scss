.landing_form {
  border: 2px solid #f9d71c;
  height: 250px;
  width: 500px;
  justify-content: space-around;
  align-items: center;
  margin-top: 20%;
}

div {
  display: flex;
  flex-direction: column;
  text-align: center;
  
  h1 {
    color: #f9d71c;
  }

  input {
    height: 30px;
    width: 250px;
    margin: 2%;
  }

  button {
    height: 25px;
    margin: 2%;
    width: 200px;
    border: 1px solid #f9d71c;
    background: black;
    color: #f9d71c;
  }

  .rank-menu {
    width: 40%;
  }
}