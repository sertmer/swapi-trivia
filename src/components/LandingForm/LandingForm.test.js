import React from 'react'
import LandingForm from './LandingForm'
import { shallow } from 'enzyme'

describe('LandingForm', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<LandingForm />)
  })

  it('should match the snapshot', () => {   
    expect(wrapper).toMatchSnapshot()
  })

})