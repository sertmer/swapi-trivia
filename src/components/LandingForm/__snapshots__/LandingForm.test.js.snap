// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingForm should match the snapshot 1`] = `
<div
  className="landing_form"
>
  <h1>
    Star Wars Trivia Database
  </h1>
  <input
    className="form_name"
    name="name"
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    className="form_quote"
    name="quote"
    onChange={[Function]}
    placeholder="Favorite Star Wars Quote"
    type="text"
    value=""
  />
  <select
    className="rank-menu"
    name="rank"
    onChange={[Function]}
    value=""
  >
    <option
      defaultValue="rank"
    >
      Select Your Rank..
    </option>
    <option
      name="rank"
      value="Padawan"
    >
      Padawan
    </option>
    <option
      name="rank"
      value="Jedi Knight"
    >
      Jedi Knight
    </option>
    <option
      name="rank"
      value="Jedi Master"
    >
      Jedi Master
    </option>
  </select>
  <button
    className="form_button"
    disabled={true}
  >
    Submit
  </button>
</div>
`;
